group-coordinator/src/main/java/org/apache/kafka/coordinator/group/modern/ModernGroup.java

```diff
@@ -82,7 +81,7 @@ public static class DeadlineAndEpoch {
     /**
      * The number of subscribers or regular expressions per topic.
      */
-    protected final TimelineHashMap<String, Integer> subscribedTopicNames;
+    protected final TimelineHashMap<String, SubscriptionCount> subscribedTopicNames;
```

Comment:
I changed how we bookkeep the subscription count to differentiate regex subs and names subs.